include/nuttx/spinlock.h

```diff
@@ -532,14 +528,7 @@ irqstate_t spin_lock_irqsave_wo_note(FAR volatile spinlock_t *lock)
 
   if (NULL == lock)
     {
-      int me = this_cpu();
```

Comment:
but the change may break the current code base?

Reply:
No, I just reviewed the current code base and did not find any nested use of spinlock

Reply:
should we remove NULL pointer support directly?

Reply:
> enter_critical_section support the nest lock, spin_lock with NULL ptr mimic the critical section behavior, it's better to support the similar behavior

Nested spinlock are fictitious demand, and other operating systems do not provide this capability.which will allow kernel developers to abuse this feature (fortunately current code base does not use nested spinlock)

> or remove NULL pointer support directly.

This will be another topic

Reply:
OK, @hujun260 please take a look.

Reply:
This change is fine. Nested spinlock is indeed very harmful, prone to abuse and leading to a decline in code quality. Additionally, this functionality is not available in Linux. In our future plans, nested usage of enter_critical_section will also be removed.